### ISSUE TITLE
Invalid raw string literal error emitted when quotes are used as d-char sequence

### ISSUE THREAD

**Issue by lyrahgames** [build2/build2#268]
# Context
```
$ b --version
build2 0.16.0-a.0.53e34adc6fa8
libbutl 0.16.0-a.0.dbce2c5f2421
host x86_64-linux-gnu
Copyright (c) 2014-2023 the build2 authors.
This is free software released under the MIT license.
```

# How to reproduce?
Create a test folder with the  files `main.cpp` and `buildfile`, insert the following content, and run the command `b`.
```c++
// main.cpp
int main() { R""(test)""; }
```
```
# buildfile
using cxx
exe{main}: cxx{main.cpp}
```

# Observed
```
$ b
main.o.ii:446:15: error: invalid raw string literal
  info: while scanning cxx{main}
  info: while applying rule cxx.compile to update obje{main}
  info: while applying rule cxx.link to update exe{main}
  info: while applying rule build.alias to update dir{./}
info: failed to update dir{./}
```

# Expected
The scanning of `b` should not emit any errors.

# Notes
- According to [cppreference.com](https://en.cppreference.com/w/cpp/language/string_literal) the given raw string literal is syntactically allowed. 
- Using GCC or Clang, the code compiles without errors.
```
g++ -o main main.cpp
clang++ -o main main.cpp
```
- Removing or changing the d-char sequence from `"` to another character, like `/`, makes `b` run without errors.
```c++
// main.cpp
int main() { R"(test)"; }
```
```c++
// main.cpp
int main() { R"/(test)/"; }
```

**Comment by boris-kolpackov** [comment]
Thanks for the detailed report! Yes, there was a bug in raw string literal lexing which is now fixed for 0.16.0 and the fix is available in the latest staged toolchain.

**Comment by lyrahgames** [comment]
Perfect. Thank you for the quick fix. I have already tried the latest staged toolchain and it worked.
